main/src/main/java/org/apache/james/jdkim/DKIMCommon.java

```diff
@@ -80,15 +80,15 @@ protected static void signatureCheck(Headers h, SignatureRecord sign,
             // NOTE check this getter is case insensitive
             List<String> hl = h.getFields(header.toString());
             if (hl != null && hl.size() > 0) {
-                Integer done = processedHeader.get(header.toString());
+                Integer done = processedHeader.get(header.toString().toLowerCase());
```

Comment:
.toLowerCase() should always take a Locale (otherwize the result depends on the system setting you are running on).

If you don't get any better choice UTF-8 is the thing to go to IMO.